### PR TITLE
Graph module poc

### PR DESCRIPTION
**Pull request by harryviennot**

### CODE REVIEW

File: sources/Core/Image/Image.hpp [hunk 1]
```diff
@@ -5,25 +5,28 @@
...
+#pragma once
+
 #include <string>
 #include <vector>
 #include "Vec3.hpp"
 #include "Scene.hpp"
 
 namespace RayTracer {
-    namespace Core {
+    namespace Shared {
```
Comment (ClementLagasse) [inline sources/Core/Image/Image.hpp:17]: Dans le Core mon neuf

Comment (theohmwoa) [inline sources/Core/Image/Image.hpp:17]: Core

File: sources/Core/Image/Image.hpp [hunk 2]
```diff
@@ -5,25 +5,28 @@
...
-            void save(const std::string& filename) const;
+            std::vector<std::vector<Vec3>> getPixels() const { return pixels; };
 
         private:
-            void setPixel(int x, int y, const RayTracer::Shared::Vec3 &color);
+            void setPixel(int x, int y, const Vec3& color);
             int width;
             int height;
-            std::vector<std::vector<RayTracer::Shared::Vec3>> pixels;
+            std::vector<std::vector<Vec3>>  pixels;
```
Comment (theohmwoa) [inline sources/Core/Image/Image.hpp:29]: RayTracer::Shared::

File: sources/Plugins/GraphModule/POC/PocEntryPoint.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+** EPITECH PROJECT, 2023
+** raytracer
+** File description:
+** PocEntryPoint
+*/
+
+#include "PocGraphModule.hpp"
+
+extern "C" {
+    enum LibType { GRAPH = 0, GAME = 1, HOMESCREEN = 2};
```
Comment (ClementLagasse) [inline sources/Plugins/GraphModule/POC/PocEntryPoint.cpp:11]: Entity, GraphModule, Decorator

File: sources/Plugins/GraphModule/POC/PocGraphModule.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            auto pixels = image.getPixels();
+            if (!outputFile.is_open()) {
+                std::cerr << "Error: Unable to open the output file: " << filename << std::endl;
+                eturn;
+            }
+            outputFile << "P3\n" << width << " " << height << "\n255\n";
+
+            for (int y = 0; y < height; ++y) {
+                for (int x = 0; x < width; ++x) {
+                    const Vec3& color = pixels[y * width + x];
```
Comment (theohmwoa) [inline sources/Plugins/GraphModule/POC/PocGraphModule.cpp:32]: Faux ??? Double vector

**Review by ClementLagasse — APPROVED** [review]

**Review by theohmwoa — APPROVED** [review]

**Pull request merged**